### PR TITLE
Feature/update actor v7

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
### Description of the Change
Update for Snap-deals:
- ExtendedSectorInfo
- API changes
- Actor V7 data model

### CODE REVIEW

File: core/const.cpp [hunk 1]
```diff
@@ -32,27 +32,35 @@ namespace fc {
   DEFINE(kUpgradeSmokeHeight){51000};
   DEFINE(kUpgradeIgnitionHeight){94000};
   DEFINE(kUpgradeRefuelHeight){130800};
-  DEFINE(kUpgradeActorsV2Height){138720};
+  DEFINE(kUpgradeAssemblyHeight){138720};
```
Comment (turuslan) [inline core/const.cpp:35]:
```suggestion
  DEFINE(kUpgradeAssemblyHeight){138720}; // actors v2
```

File: core/const.cpp [hunk 2]
```diff
@@ -151,17 +161,18 @@ namespace fc {
     kUpgradeIgnitionHeight = 100000000;
```
Comment (turuslan) [inline core/const.cpp:161]:
```suggestion
    kUpgradeIgnitionHeight = INT64_MAX;
```

File: core/const.cpp [hunk 3]
```diff
@@ -193,18 +204,20 @@ namespace fc {
...
+    kUpgradeCalicoHeight = -9;
+    kUpgradePersianHeight = -10;
+    kUpgradeOrangeHeight = -11;
+    kUpgradeClausHeight = -12;
+    kUpgradeTrustHeight = -13;
+    kUpgradeNorwegianHeight = -14;
+    kUpgradeTurboHeight = -15;
+    kUpgradeHyperdriveHeight = -16;
+    kUpgradeChocolateHeight = -17;
+    kUpgradeOhSnapHeight = -18;
```
Comment (turuslan) [inline core/const.cpp:220]:
```suggestion
    kUpgradeChocolateHeight = INT64_MAX;
    kUpgradeOhSnapHeight = INT64_MAX;
```

File: core/vm/toolchain/impl/toolchain.cpp [hunk 4]
```diff
@@ -111,6 +111,18 @@ namespace fc::vm::toolchain {
...
+        || (actorCid == v7::kStorageMarketCodeId)
+        || (actorCid == v7::kStorageMinerCodeId)
+        || (actorCid == v7::kMultisigCodeId) || (actorCid == v7::kInitCodeId)
+        || (actorCid == v7::kPaymentChannelCodeId)
+        || (actorCid == v7::kRewardActorCodeId)
+        || (actorCid == v7::kSystemActorCodeId)
+        || (actorCid == v7::kVerifiedRegistryCodeId)) {
+      return ActorVersion::kVersion7;
+    }
+
```
Comment (turuslan) [inline core/vm/toolchain/impl/toolchain.cpp:125]: may wrap in macro

File: test/core/blockchain/block_validator_test.cpp [hunk 5]
```diff
@@ -29,6 +29,10 @@ namespace fc::blockchain::block_validator {
     ProofParamProvider::getParams(params, 0).value();
 
     setParamsInteropnet();
+    // Set network version 13
+    kUpgradeChocolateHeight = INT64_MAX;
+    kUpgradeOhSnapHeight = INT64_MAX;
+
```
Comment (turuslan) [inline test/core/blockchain/block_validator_test.cpp:35]:
```suggestion
```

File: core/const.hpp [hunk 6]
```diff
@@ -77,16 +78,21 @@ namespace fc {
...
   extern ChainEpoch kUpgradeNorwegianHeight;
+  /** Actors V4 upgrade */
   // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
-  extern ChainEpoch kUpgradeActorsV4Height;
+  extern ChainEpoch kUpgradeTurboHeight;
   // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
   extern ChainEpoch kUpgradeHyperdriveHeight;
   // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
   extern ChainEpoch kUpgradeChocolateHeight;
+  /** Update introduces Snap deals */
```
Comment (turuslan) [inline core/const.hpp:93]:
```suggestion
  /** Actors V7 upgrade, Snap deals */
```

File: core/const.cpp (hunk 2 above)
Comment (Alexey-N-Chernyshov) [reply core/const.cpp]: Please, check if I get your comment correctly.

File: core/sector_storage/impl/manager_impl.cpp [hunk 7]
```diff
@@ -311,12 +314,22 @@ namespace fc::sector_storage {
...
-                        static_cast<SectorFileType>(SectorFileType::FTCache
-                                                    | SectorFileType::FTSealed),
-                        SectorFileType::FTNone,
-                        PathType::kStorage);
+      SectorFileType sector_file_type = SectorFileType::FTNone;
+      if (sector.sector_key.has_value()) {
+        logger_->debug("Posting over updated sector for sector id: {}",
+                       sector.sector);
+        sector_file_type = static_cast<SectorFileType>(
+            SectorFileType::FTCache | SectorFileType::FTUpdate);
```
Comment (turuslan) [inline core/sector_storage/impl/manager_impl.cpp:322]:
```suggestion
            SectorFileType::FTUpdateCache | SectorFileType::FTUpdate);
```

File: core/vm/actor/codes.hpp [hunk 8]
```diff
@@ -7,100 +7,41 @@
...
-    constexpr auto kStorageMarketCodeId{code::market##V};      \
-    constexpr auto kStorageMinerCodeId{code::miner##V};        \
-    constexpr auto kMultisigCodeId{code::multisig##V};         \
-    constexpr auto kInitCodeId{code::init##V};                 \
-    constexpr auto kPaymentChannelCodeId{code::paych##V};      \
-    constexpr auto kRewardActorCodeId{code::reward##V};        \
-    constexpr auto kSystemActorCodeId{code::system##V};        \
-    constexpr auto kVerifiedRegistryCodeId{code::verifreg##V}; \
+namespace fc::vm::actor::builtin::v0 {
+  constexpr ActorCodeCid kAccountCodeId{"fil/1/account"};
```
Comment (turuslan) [inline core/vm/actor/codes.hpp:11]: `VM_ACTOR_BUILTIN_V_CODE` can accept two params (namespace and string)

**Review by turuslan — APPROVED** [review]

File: core/primitives/sector/sector.hpp [hunk 9]
```diff
@@ -174,6 +174,28 @@ namespace fc::primitives::sector {
            && lhs.sector == rhs.sector && lhs.sealed_cid == rhs.sealed_cid;
   }
 
+  struct ExtendedSectorInfo {
```
Comment (wer1st) [inline core/primitives/sector/sector.hpp:177]: Necessary to apply another approach. We don't need to introduce new struct here

Comment (wer1st) [reply core/primitives/sector/sector.hpp:177]: Discussed in PM

**Review by wer1st — APPROVED** [review]

**Pull request merged**